airflow/utils/dag_processing.py

```diff
@@ -288,18 +288,21 @@ def correct_maybe_zipped(fileloc):
 COMMENT_PATTERN = re.compile(r"\s*#.*")
 
 
-def list_py_file_paths(directory, safe_mode=True,
+def list_py_file_paths(directory, safe_mode=None,
```

Comment:
```suggestion
def list_py_file_paths(directory, safe_mode=conf.getboolean('core', 'a', fallback=False)
```

This one-liner should work.

Can you also add some tests so that we can discover this kind of issue in future.

Reply:
Hi @kaxil,

I'm happy to add tests, but as mentioned in the PR description, I need some pointers. Specifically:
- which of the two testing strategies that I suggest would be most appropriate
- if the second, I'm going to have to refactor things, and I don't want to do that if it's going to be in any way controversial
- is there a context manager (or something similar) for temporarily setting configuration variables

Thanks,
jml

Reply:
Changed to set `safe_mode` via default arguments, although I'm falling back to `True` rather than `False`, since that's what the behaviour was before the option was introduced.

Note that this means that configuration changes within the lifetime of the process will not be respected by the default argument, which might surprise people.

Reply:
Other than in tests, there is no way to change the config of a running Airflow process, so that isn't a problem.

We have `from tests.test_utils.config import config_vars` which is the decorator you change configs for a test (but this won't change the default value here as you are aware.)

I think the way I would approach this is to _not_ read the config setting here (i.e. no default), but instead pass the value down the scheduler job, and test via mocking that we pass the value along, which probably means adding it as attributes to a number of places in the call stack. Hmmmm. Maybe leaving this here _is_ fine. (I've been off for a week, brain still isn't firing on all cylinders yet)

(The _behaviour_ of `list_py_file_paths` is already tested via tests/models/test_dagbag.py)

Reply:
I agree with your suggested approach: in other projects I have it as a rule that only the very outer-most layers touch the configuration systems, and everything else is passed as parameters.

The reason I chose this approach is that it's the same approach that's already used for `include_examples`. I thought having two different approaches would be more confusing.

Relatedly, I also think it's a bad idea to have a default value for this parameter, but I'm not sure what the backwards compatibility policies are. I guess I could create a new function with mandatory params. 

I'm honestly happy to roll up my sleeves and do a chunk of work, I just don't want to get caught in interminable discussion. We've already collectively spent a lot of words & roundtrips on a one line change (my first Airflow PR!), and I'm wary that a 600 line change would be much worse. 

So, what do I need to do to get this merged?

Reply:
Since this is such an internal method that no-one outside of the scheduler is likely to call I'm not very concerned about breaking changes to the sig of this function.

Reply:
I've re-run the failing test (we haven't managed to fix all flaky tests yet 😢), if that goes green we're probably good to merge.